### ISSUE TITLE
Craft frontend debug toolbar prevents rendering of tags

### ISSUE THREAD

**Issue by evanwarner** [nystudio107/craft-seomatic#106]
Having the front end debug toolbar enabled on my user account prevents the core SEOmatic tags from rendering, and logs these types of errors:
```
nystudio107\seomatic\models\MetaTag::prepForRender

Meta tag `description` did not render because it is missing attributes. Array
(
    [name] => description
)
```

My general setup is that I have *SEO Title*, *SEO Description*, and *Social Sharing Image* fields on all my entries, and in Content SEO settings each section is set to get SEO data from those fields. The fields on each entry definitely have content, and as soon as I turn the debug toolbar off, they all render correctly.

If you can't reproduce this let me know and I'll provide more detailed information about my config.

**Comment by khalwat** [comment]
I'll have a look at this. Very curious.

**Comment by khalwat** [comment]
So I actually _can't_ reproduce this @evanwarner -- I have the Yii2 debug toolbar enabled on the frontend, and if I choose "View Source" it is indeed outputting what I'd expect.

Can you help me reproduce this?

**Comment by evanwarner** [comment]
I cleared my Craft logs, loaded up a page once with the toolbar on and once with it off. Then I diffed the log files and noticed a definite relevant difference. This difference exists for each attempt to render a `title` or `description`. (Not sure what the extra two lines with different string template ids are in the first chunk.)

Toolbar on (tags don't render):
```
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][profile begin][craft\web\twig\Template::display] __string_template__4566f149ce77a81413cf68a84844af2d0b07002dd034b065ba4de9eba4006094
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][profile end][craft\web\twig\Template::display] __string_template__4566f149ce77a81413cf68a84844af2d0b07002dd034b065ba4de9eba4006094
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][profile begin][craft\web\twig\Template::display] __string_template__1afd5b259028971eeda70607d4326663a96353121b823e1fb55056424ec65702
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][profile end][craft\web\twig\Template::display] __string_template__1afd5b259028971eeda70607d4326663a96353121b823e1fb55056424ec65702
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][error][nystudio107\seomatic\models\MetaTag::prepForRender] Meta tag `description` did not render because it is missing attributes. Array
(
    [name] => description
)
```

Toolbar off (tags render fine):
```
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][profile begin][craft\web\twig\Template::display] __string_template__4566f149ce77a81413cf68a84844af2d0b07002dd034b065ba4de9eba4006094
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][profile end][craft\web\twig\Template::display] __string_template__4566f149ce77a81413cf68a84844af2d0b07002dd034b065ba4de9eba4006094
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][info][nystudio107\seomatic\base\MetaItem::debugMetaItem] Scenario: "warning"
<meta name="description" content="Testing a description field">
(timestamp): [::1][1][3c35dbfdf27dd66b6dc826b31651382a][warning][nystudio107\seomatic\base\MetaItem::debugMetaItem] Tag attribute: content -> Content should contain at least 70 characters.
```

Other observations:

1. Regardless of the debug bar status, the other meta fields that are indeed empty _always_ log the first format (but with only one of the `profile begin`/`profile end`sequences before them).
1. Strangely (and maybe this is important), even when I input well over 70 characters for a description (or 40 for a title), the too few characters warning shows up in the log.

**Comment by evanwarner** [comment]
Any particular things I can provide that might help you try to recreate this?

**Comment by khalwat** [comment]
Well, the question I have is whether or not when you view the source to the page if the tags are actually there or not?

**Comment by evanwarner** [comment]
If the toolbar is on, the title and description meta tags (including `og:` and `twitter:`) do not appear in source. And the HTML title tag renders as so:

`<title>  | SiteName</title>`

(where `SiteName` is the correct client name in my build)

**Comment by khalwat** [comment]
So I've tried this on two different computers/local dev environments, both running:

Craft CMS 3.0.8
SEOmatic 3.0.13

I'm seeing all of the tags render, both in the DOM, and when I do "View Source" from Chrome 66

Can you give me the details of the environment where you're able to reproduce this?  Also can you see if you're able to reproduce this in just a very generic/empty template?

You can use just:

```
<!DOCTYPE html>
<html>
<head>
</head>
<body>

{{ dump(seomatic.meta) }}

</body>
</html>
```

**Comment by evanwarner** [comment]
Alright, first I updated SEOmatic to 3.0.13 (from 3.0.11), and the fallbacks kicked in. Instead of rendering nothing when the toolbar was on, it would render the global SEO.

Then I updated Craft to 3.0.8 (I was at 3.0.6), and the problem is gone! Toolbar on or off, the appropriate meta tags are rendering with correct info.

Whether it was actually something that changed in the code or my instance just needing a kick in the ass, I'm not sure. But I guess we can close this one and I'll hope it doesn't surface again.
